src/main/webapp/dev/js/common/questionRank.js

```diff
@@ -235,6 +235,28 @@ function hideInvalidRankRecipientFeedbackPaths(qnNum) {
     }
 }
 
+/**
+ * Auto fills the min and max options to 1 if empty when the corresponding checkbox is checked
+ * in Rank options and Rank recipients questions
+ */
+function bindAutoFillEmptyRankOptionsChangeEvent() {
+    $(`[id^=${ParamsNames.FEEDBACK_QUESTION_RANK_IS_MAX_OPTIONS_TO_BE_RANKED_ENABLED}]`
+            + `, [id^=${ParamsNames.FEEDBACK_QUESTION_RANK_IS_MIN_OPTIONS_TO_BE_RANKED_ENABLED}]`
+            + `, [id^=${ParamsNames.FEEDBACK_QUESTION_RANK_IS_MIN_RECIPIENTS_TO_BE_RANKED_ENABLED}]`
+            + `, [id^=${ParamsNames.FEEDBACK_QUESTION_RANK_IS_MAX_RECIPIENTS_TO_BE_RANKED_ENABLED}]`).change((e) => {
+        const linkedInputId = $(e.currentTarget).data('linkedInputId');
+        const linkedInput = $(`#${linkedInputId}`);
+
+        if ($(e.currentTarget).prop('checked')) {
+            if ($(linkedInput).val() === '') {
+                $(linkedInput).val('1');
+            }
+            // TODO: Add 'required' to HTML file instead, but redesign form to exclude other questions. See issue #8688.
+            $(linkedInput).prop('required', true); // Add 'required' to input for HTML validation when clicked on checkbox
```

Comment:
@Crphang As I am working on a similar issue, and was referring this PR, Wanted to discuss something. And that is,

Consider this scenario,
- Add a rank option/recipient question
- check the max/min option. (Now, the field is required)
- Uncheck the max/min option (The field is still required? as we only consider when the checkbox is checked?)
- Cancel the question
- Add another question, like an essay question
- Submit on google chrome, (We should get `Invalid form control` error, as the max/min option is hidden and still required)

So, In this case, isn't this work around incomplete.

Reply:
Thanks for raising this. I tried it and it does fail. To make it complete would be to add another handler on the cancel add question button, and remove required for each field that needs to be removed. 

I am also thinking of adding a `{isRequired}` tag in the HTML and to be rendered `"required"` only when it is the shown question or `""` when it is not. I will try both and write the results in the issue.

Reply:
@Crphang 
There is one more scenario where it will fail which I forgot to mention,
- Add a rank opiton/recipient/ MSQ question
- Check the max/min option (Now, it is required)
- Cancel the question
- Add another question
- Submit (It will fail, as even though we canceled the question, the max/min option is still checked, so the field is still required)

I implemented something like, in the `InstructorFeedbackEdit.js`, in `enableNewQuestion()` function, when we add a new question, it will check if the max/min field (for MSQ) is hidden, if it is, it will set required as false.

>To make it complete would be to add another handler on the cancel add question button, and remove required for each field that needs to be removed.

But I guess this (or the other option you mentioned) will be a better solution than what I have implemented.

Reply:
@sukanta-27 I think that will work as well. Then it can save us on adding the handlers. I will let you know about the other method.

Reply:
@sukanta-27 If you can, document your method in the issue #8688. So that all related PRs can have a consistent workaround temporary solution after we agree on it.

Reply:
@Crphang Sure. Will do.